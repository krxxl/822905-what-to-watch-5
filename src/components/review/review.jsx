import React from 'react';
import Header from '../header/header';
import PropTypes from 'prop-types';

const Review = (props) => {
  const {films, filmId} = props;
  const film = films.find((item) => item.id === +filmId);
  const {hero, name, poster} = film;
  return (
    <section className="movie-card movie-card--full">
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src={hero} alt={name} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <Header />

        <div className="movie-card__poster movie-card__poster--small">
          <img src={poster} alt={name} width="218" height="327" />
        </div>
      </div>

      <div className="add-review">
        {props.children}
      </div>

    </section>
  );
};

Review.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    hero: PropTypes.string.isRequired,
    time: PropTypes.string.isRequired,
    rating: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    desc: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.array.isRequired,
    isInList: PropTypes.bool.isRequired,
    video: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  }).isRequired).isRequired,
  filmId: PropTypes.number.isRequired,
  children: PropTypes.element.isRequired,
};

export default Review;
