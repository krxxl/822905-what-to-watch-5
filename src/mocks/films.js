export default [{
  name: `The Grand Budapest Hotel`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  preview: `img/bohemian-rhapsody.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 1,
},
{
  name: `bohemian-rhapsody`,
  genre: `Sci-Fi`,
  year: `2010`,
  poster: `img/bohemian-rhapsody.jpg`,
  preview: `img/bohemian-rhapsody.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `2h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 2,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 3,
}, {
  name: `fantastic-beasts-the-crimes-of-grindelwald`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 15m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 4,
}, {
  name: `johnny-english`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/johnny-english.jpg`,
  preview: `img/johnny-english.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 5,
}, {
  name: `macbeth`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/macbeth.jpg`,
  preview: `img/macbeth.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 6,
}, {
  name: `midnight-special`,
  genre: `kids & family`,
  year: `2010`,
  poster: `img/midnight-special.jpg`,
  preview: `img/midnight-special.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 7,
}, {
  name: `moonrise-kingdom`,
  genre: `thriller`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 8,
}, {
  name: `moonrise-kingdom`,
  genre: `horror`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 9,
}, {
  name: `moonrise-kingdom`,
  genre: `documentary`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 10,
}, {
  name: `moonrise-kingdom`,
  genre: `drama`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 11,
}, {
  name: `moonrise-kingdom`,
  genre: `Sci-Fi`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 12,
}, {
  name: `moonrise-kingdom`,
  genre: `Romance`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 13,
}, {
  name: `moonrise-kingdom`,
  genre: `Kids & Family`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 14,
}, {
  name: `moonrise-kingdom`,
  genre: `Horror`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 15,
}, {
  name: `moonrise-kingdom`,
  genre: `Drama`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 16,
}, {
  name: `moonrise-kingdom`,
  genre: `Documentary`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 17,
}, {
  name: `moonrise-kingdom`,
  genre: `Crime`,
  year: `2010`,
  poster: `img/moonrise-kingdom.jpg`,
  preview: `img/moonrise-kingdom.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 39m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  isInList: true,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  id: 18,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 19,
}, {
  name: `dardjeeling-limited`,
  genre: `comedy`,
  year: `2010`,
  poster: `img/dardjeeling-limited.jpg`,
  preview: `img/dardjeeling-limited.jpg`,
  hero: `img/bg-the-grand-budapest-hotel.jpg`,
  time: `1h 01m`,
  rating: `9.8`,
  count: 240,
  desc: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustaves friend and protege.  Gustave prides himself on providing first-class service to the hotels guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustaves lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`],
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isInList: true,
  id: 20,
},
];
