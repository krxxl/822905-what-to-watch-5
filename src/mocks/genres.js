export const genres = [
  {
    name: `All genres`,
    id: 1
  },
  {
    name: `Comedies`,
    id: 2
  },
  {
    name: `Crime`,
    id: 3
  },
  {
    name: `Documentary`,
    id: 4
  },
  {
    name: `Dramas`,
    id: 5
  },
  {
    name: `Horror`,
    id: 6
  },
  {
    name: `Kids & Family`,
    id: 7
  },
  {
    name: `Romance`,
    id: 8
  },
  {
    name: `Sci-Fi`,
    id: 9
  },
  {
    name: `Thrillers`,
    id: 10
  },
];

export const genresInSingular = {
  ALL: `All genres`,
  COMEDY: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMA: `Dramas`,
  HORROR: `Horror`,
  KIDS: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLER: `Thrillers`,
};
