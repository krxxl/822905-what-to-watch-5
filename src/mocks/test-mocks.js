import films from './films';
import reviews from './reviews';
import {AuthorizationStatus} from "../constant/constant";
import {SHOW_ON_STAR_FILMS} from '../constant/constant';

export const TEST_MOCK_STORE = {
  DATA: {
    films,
    reviews,
    favoriteFilms: films,
    isLoading: true,
    isPromoLoading: true,
    sendingReview: false,
    sendingReviewError: false,
    promoFilm: films[0],
  },
  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
  SHOW: {
    genreActive: `All genres`,
    count: SHOW_ON_STAR_FILMS,
  }
};
