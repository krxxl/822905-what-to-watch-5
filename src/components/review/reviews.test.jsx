import React from 'react';
import renderer from 'react-test-renderer';
import {Review} from './review';
import films from '../../mocks/films';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import {TEST_MOCK_STORE} from '../../mocks/test-mocks';
import thunk from 'redux-thunk';
import {createAPI} from '../../services/api';
import {MemoryRouter} from 'react-router-dom';

const api = createAPI(() => {});

const middlewares = [thunk.withExtraArgument(api)];
const mockStore = configureMockStore(middlewares);
const store = mockStore(TEST_MOCK_STORE);

describe(`Render Review`, () => {
  it(`Render Review`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <Review
                id={films[0].id}
                film={films[0]}
                name={films[0].name}
                posterImage={films[0].posterImage}
                backgroundImage={films[0].backgroundImage}
                filmId={films[0].id}>
                <form></form>
              </Review>
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
