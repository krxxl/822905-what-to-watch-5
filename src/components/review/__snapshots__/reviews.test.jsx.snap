// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Review Render Review 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head "
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form />
  </div>
</section>
`;
